manifests/params.pp

```diff
@@ -29,9 +29,18 @@
   if $::operatingsystem == 'Ubuntu' {
     if versioncmp($::lsbdistrelease, '8.04') < 1 {
       $init_style = 'debian'
-    } else {
+    } elsif versioncmp($::operatingsystemrelease, '15.04') < 0 {
       $init_style = 'upstart'
     }
+    else {
```

Comment:
It's a thought.  I know at our University we have a computing science department with a good amount of 15.04s running upstart, but with plans to move them to systemd once they get a better understanding of it and integrate it.  We're still on puppet 3 with plans to jump into puppet 4 soon enough.  Puppet 3.x doesn't support fedora 22 or Ubuntu 15.04 so I wrote the is_systemd fact specifically for Ubuntu 15.04 so I could put that into my sites.pp file to avoid it thinking that it was upstart still.  Fedora 22 didn't need the fact but still needed to get a Service { provider => 'systemd' } due to it not being in puppet 3.x

The downside to doing overrides on the init_style is that you can't have a mixture of upstart and systemd Ubuntu 15.04 in the same role/profile or hostgroup(foreman)  Otherwise you'd have to put in a few more snowflakes.